File: IDATT2106 Systemutvikling 2 med smidig prosjekt/backend/src/main/java/com/ntnu/gidd/dto/User/UserRegistrationDto.java
package com.ntnu.gidd.dto.User;

import com.ntnu.gidd.validation.FieldMatch;
import com.ntnu.gidd.validation.ValidPassword;
import lombok.*;

import javax.validation.constraints.Email;
import javax.validation.constraints.NotEmpty;
import javax.validation.constraints.NotNull;
import java.time.LocalDate;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class UserRegistrationDto {
      @NotNull
      @NotEmpty
      private String firstName;

      @NotNull
      @NotEmpty
      private String surname;

      
      @NotNull
      @NotEmpty
      @ValidPassword
      private String password;

      @NotNull
      @NotEmpty
      @Email
      private String email;

      private LocalDate birthDate;
}

File: IDATT2105 Full-stack applikasjonsutvikling/Rombestilling/backend/src/main/kotlin/ntnu/idatt2105/user/dto/UserRegistrationDto.kt
package ntnu.idatt2105.user.dto

import com.opencsv.bean.CsvBindByName
import java.time.LocalDate
import javax.validation.constraints.Email
import javax.validation.constraints.NotBlank

data class UserRegistrationDto(
    @get:NotBlank(message = "Field must not be blank")
    @CsvBindByName(column = "firstName")
    val firstName: String,
    @get:NotBlank(message = "Field must not be blank")
    @CsvBindByName(column = "surname")
    val surname: String,
    @get:NotBlank(message = "Field must not be blank")
    @get:Email(message = "Please provide a valid email")
    @CsvBindByName(column = "email")
    val email: String,
    @get:NotBlank(message = "Field must not be blank")
    @CsvBindByName(column = "phoneNumber")
    val phoneNumber: String,
    var expirationDate: LocalDate = LocalDate.now().plusYears(1),
) {
    constructor() : this("", "", "", "",)
}
